src/software/new_geom/util/intersects.cpp

```diff
@@ -73,13 +67,12 @@ bool intersects(const Circle &first, const Circle &second)
 
 bool intersects(const Segment &first, const Circle &second)
 {
-    bool segment_endpoint_inside_circle =
-        second.contains(first.getSegStart()) || second.contains(first.getEnd());
-
-    bool segment_crosses_circle =
-        distance(first, second.getOrigin()) < second.getRadius();
+    if (distance(first, second.getOrigin()) < second.getRadius())
+    {
+        return true;
+    }
 
-    return segment_endpoint_inside_circle || segment_crosses_circle;
+    return false;
```

Comment:
Meta discussion here: @garethellis0 @MathewMacDougall @akhilveeraghanta 

This wouldn't count where a segment is 'just on' the circle as an intersection. I *feel* like that should count? To me an intersection is different than contains.

Also any segment that exists entirely within the circle counts as intersecting, is that what we want?

Reply:
![Sketch 2](https://user-images.githubusercontent.com/25271038/84187142-75435100-aa46-11ea-9cb7-f5be7151747c.png)

Also, I don't think this case is handled?

Reply:
This case in the drawing is handled because the distance from end to origin is less than the radius.

I can change the check to distance<=radius so that it would include "just on" intersections

Reply:
Ohhh, OK I see that now 👍